### ISSUE TITLE
paperless unpaper performing worse that stock unpaper

### ISSUE THREAD

**Issue by towo** [danielquinn/paperless#189]
My scanner has the unfortunate tendency to produce slightly skewed files, so I was hoping that unpaper would help here; set up paperless, consuming works fine, but unpaper doesn't work as good as I thought.

Then I manually ran unpaper against the file, and it produces a way better result.

Steps to reproduce for that:

```
$ pdftoppm scan.pdf > scan.ppm
$ unpaper scan.ppm processed.ppm
$ convert processed.ppm processed.pdf
```

Here's the file as paperless has it consumed from scan.pdf:

![screenshot from 2017-02-09 13-53-48](https://cloud.githubusercontent.com/assets/271819/22784176/55cc8cf0-eecf-11e6-8528-436a7483234a.png)

And here's my processed.pdf:

![screenshot from 2017-02-09 13-54-13](https://cloud.githubusercontent.com/assets/271819/22784183/5d908630-eecf-11e6-9207-34551b1f2e2a.png)

**Comment by danielquinn** [comment]
Ah I think I understand what's going on here.  It took me a while and went digging through the code to see if we were invoking unpaper in some special way.  I think what you're running into is how Paperless doesn't modify your original PDF in any way.  The process is like this:

1. The original PDF is found and converted into a thumbnail (imagemagick) and a greyscale `.pnm`.  The latter of which has unpaper used on it.
2. Tesseract reads the `.pnm` to generate the OCR text which is then dumped into the database.
3. The `.pnm` is then deleted and *the original PDF is stored, encrypted by Paperless*.

In other words, Paperless only uses unpaper to get a better OCR result, but the PDF it stores the same one you gave it.  If you want to tweak your PDFs, you should do so *before* pushing it into Paperless.

I hope that explains things, but if you need further clarification, feel free to re-open this issue.

**Comment by towo** [comment]
Ah, thanks for the info. The documentation led me to believe it saves them deskewed, as well.
